softwareComponents/simplesimClientLib/src/simplesim_client.cpp

```diff
@@ -242,15 +242,14 @@ std::map< rofi::configuration::ModuleId, ModuleRenderInfo > addConfigurationToRe
                          moduleRenderInfos,
                          newConfiguration );
 
-    for ( const auto & moduleInfo : newConfiguration.modules() ) {
-        assert( moduleInfo.module.get() );
-        assert( moduleInfo.absPosition && "The configuration has to be prepared" );
-        moduleRenderInfos[ moduleInfo.module->getId() ].componentActors =
+    for ( const auto & moduleWithPos : newConfiguration.modulesWithAbsPos() ) {
+        const rofi::configuration::Module & mod = moduleWithPos.first;
```

Comment:
Dvojice z `modulesWithAbsPos()` by šlo rozbalit přímo; u modulu je typ poměrně zřejmý, a absPosition jej nemá specifikovaný, tedy by nedošlo ke ztrátě informace.
Namísto `mod` je možná i tady lepší `rModule`, pokud tedy chceme změnit názvy pro všechny (relevantní) proměnné typu Module na `rModule` (pak by bylo i méně pravděpodobné si splést typ pokud bychom přímo rozbalovali dvojici).

Reply:
S tím názvem máš pravdu - to jsem přehlídl.
Nicméně s tím rozbalením je to horší :/ abych vrátil něco ve dvojici (`std::pair`), tak to musím zabalit do `std::reference_wrapper`, takže když to rozbalím přímo, tak to bude mít špatný typ...

Reply:
Tak moje chyba - měl jsem za to, že `std::pair` se v tomto chová stejně jako `std::optional`. (`std::optional` nemůže držet referenci, kdežto `std::pair` ano)